docs/product-manuals/zeebe/open-source/exporters.md

```diff
@@ -47,87 +44,69 @@ all exporters interact in the same way with the broker, which is defined by the
 
-This stream processor will create exactly one instance of each configured exporter,
-and forward every record written on the stream to each of these in turn.
+This stream processor creates exactly one instance of each configured exporter,
+and forwards every record written on the stream to each of these in turn.
 
-> **Note:** this implies that there will be exactly one instance of every exporter for
-> every partition: if you have 4 partitions, and at least 4 threads for processing,
-> then there are potentially 4 instances of your exporter exporting simultaneously.
+:::note
+This implies there will be exactly one instance of every exporter for every partition. If you have four partitions, and at least four threads for processing, there are potentially four instances of your exporter exporting simultaneously.
+:::
 
-Note that Zeebe only guarantees at-least-once semantics, that is, a record will be
-seen at least once by an exporter, and maybe more. Cases where this may happen
-include:
+Zeebe only guarantees at-least-once semantics. A record is seen at least once by an exporter, and maybe more. Cases where this may happen include the following:
 
 - During reprocessing after raft failover (i.e. new leader election)
-- On error if the position has not been updated yet
+- On error if the position is not yet updated
 
 To reduce the amount of duplicate records an exporter will process, the stream
-processor will keep track of the position of the last successfully exported record
-for every single exporter; the position is sufficient since a stream is an ordered
+processor keeps track of the position of the last successfully exported record
+for every single exporter. The position is sufficient since a stream is an ordered
 sequence of records whose position is monotonically increasing. This position is
 set by the exporter itself once it can guarantee a record has been successfully
 updated.
 
-> **Note:** although Zeebe tries to reduce the amount of duplicate records an
-> exporter has to handle, it is likely that it will have to; therefore, it is
-> necessary that export operations be idempotent.
->
-> This can be implemented either in the exporter itself, but if it exports to an
-> external system, it is recommended that you perform deduplication there to reduce
-> the load on Zeebe itself. Refer to the exporter specific documentation for how
-> this is meant to be achieved.
+:::note
+Although Zeebe tries to reduce the amount of duplicate records an exporter has to handle, it is likely that it will have to. Therefore, it is necessary that export operations be idempotent. This can be implemented either in the exporter itself, but if it exports to an external system, it is recommended you perform deduplication there to reduce the load on Zeebe itself. Refer to the exporter-specific documentation for how this is meant to be achieved.
```

Comment:
I find the second half of this sentence hard to understand: "Although Zeebe tries to reduce the amount of duplicate records an exporter has to handle, it is likely that it will have to."

Reply:
How's the following:

"Zeebe tries to reduce the amount of duplicate records an exporter handles, but it is helpful to ensure export operations are idempotent."

Reply:
I feel both are not ideal. Exporters don't **have** to be idempotent. It is just a convenient shortcut to describe how they **should** operate.

Maybe we would be better served to describe the contract that an exporter has to follow.

Something like (tech jargon, not a suggestion how to phrase it): 
- exporters are called with at least once semantics for the records to be exported
- this means they might be called more than once for any given record
- deduplication can be performed based on the record key, which is unique

Also not sure what we mean by "exporter-specific documentation". I searched the documentation for a helpful section, but couldn't find it. If it is there, we should link to it. If not, I think we should remove this reference.

Reply:
Here's what I've updated to, does it make sense to you or are there pieces you would add/remove?

"It's best to reduce the amount of duplicate records an exporter handles alongside Zeebe. Exporters are called at least once for the records to export, which means they may be called more than once for any given record. We recommend performing deduplication based on the unique record key to reduce the load on Zeebe."

Reply:
@saig0/@npepinpe : What do you think of the proposal? 

Personally, I am stumbling a bit over "reduce the load on Zeebe". I don't think that's the goal here, and also if deduplication is done badly it could hurt Zeebe performance. Plus, I think deduplication should rather be done on the backend.

At the same time, I feel getting it exactly right might be too complicated.

Reply:
You're right that reducing the load on Zeebe is not the goal of de-duplication. Further, this de-duplication should occur on your target system, not in Zeebe (in this sense, here it _is_ to reduce the load on Zeebe :smile:). 

So the important bits:

- Zeebe exports with at least once semantics, so you _will_ have to deal with duplicates
- De-duplication can be performed based on the position and partition ID - not all records have keys, but every record has a partition ID, and a unique position relative to that partition.

We can drop the reduce load part, and focus on communicating that it's crucial for the backend system to deal with duplicates. We could keep something to the effect of trying to do as little as possible in the exporter to reduce the load on the broker though. A note also on not blocking would be good, since blocking means blocking on the actor threads entirely, which can have a big impact on the system.

The exporter specific documentation here means the documentation for a specific exporter, e.g. Elasticsearch Exporter documentation, or the Kafka exporter documentation, etc. It's also possible some parts of our docs don't make sense in the new structure as well.